### ISSUE TITLE
Unclear error while rollback

### ISSUE THREAD

**Issue by 0oM4R** [threefoldtech/tfgrid-sdk-ts#500]
### Description

tried to deploy gw with tcp as backend `ip:22` 
and I got the error ` ContractNotExists` 

the expected error 
```bash
due to 255 failed to validate backend '10.238.2.2:22': failed to parse backend with error: parse "10.238.2.2:22": first path segment in URL cannot contain colon
```
i got the correct error after adding some delay before `applyAllExtrinsics` in `grid_client/src/high_level/twinDeploymentHandler.ts`

### Steps to reproduce

deploy a gw with  `10.238.2.2:22` as backend

### Full Logs/Alerts
```
Omar Kassem, [6/7/23 5:55 PM]
deploying gateway monNetwork on node 11'
Validation succeed
Start creating the gateway deployment with name wireguardssh
Merging workloads
Validation succeed
Deploying on node_id: 11
Check the name contract for the workload with name: wireguardssh
Lock acquired
2023-06-07 16:51:00        API/INIT: Not decorating runtime apis without matching versions: TransactionPaymentApi/2 (1 known)
phase: {"applyExtrinsic":1}, section: balances, method: Withdraw, data: ["5HHRg2nVfTXD9StGihrjHre5LhC2xVRZhBYvVqoorvsqo5F5",10225]
phase: {"applyExtrinsic":1}, section: smartContractModule, method: ContractCreated, data: [{"version":4,"state":{"created":null},"contractId":30483,"twinId":800,"contractType":{"nodeContract":{"nodeId":11,"deploymentHash":"0x3961653736636539666564396438633632396531616134373234373037396138","deploymentData":"0x7b2274797065223a2267617465776179222c226e616d65223a22776972656775617264737368222c2270726f6a6563744e616d65223a22227d","publicIps":0,"publicIpsList":[]}},"solutionProviderId":null}]
phase: {"applyExtrinsic":1}, section: utility, method: ItemCompleted, data: []
phase: {"applyExtrinsic":1}, section: smartContractModule, method: ContractCreated, data: [{"version":4,"state":{"created":null},"contractId":30484,"twinId":800,"contractType":{"nameContract":{"name":"0x776972656775617264737368"}},"solutionProviderId":null}]
phase: {"applyExtrinsic":1}, section: utility, method: ItemCompleted, data: []
phase: {"applyExtrinsic":1}, section: utility, method: BatchCompleted, data: []
phase: {"applyExtrinsic":1}, section: balances, method: Deposit, data: ["5EvbAkU2fMiWXGrAmnUTi4zdVNXTa9cpamnhakuQcSMYSnpT",10225]
phase: {"applyExtrinsic":1}, section: transactionPayment, method: TransactionFeePaid, data: ["5HHRg2nVfTXD9StGihrjHre5LhC2xVRZhBYvVqoorvsqo5F5",10225,0]
phase: {"applyExtrinsic":1}, section: system, method: ExtrinsicSuccess, data: [{"weight":{"refTime":2157129688,"proofSize":0},"class":"Normal","paysFee":"Yes"}]
Lock released
Sending deployment to node_id: 11
Rolling back deployments
Deleting name contract with name: wireguardssh and id: 30484
Deleting contract id 30483
Deleting contract id 30484
Lock acquired
2023-06-07 16:51:06        API/INIT: Not decorating runtime apis without matching versions: TransactionPaymentApi/2 (1 known)
phase: {"applyExtrinsic":1}, section: balances, method: Withdraw, data: ["5HHRg2nVfTXD9StGihrjHre5LhC2xVRZhBYvVqoorvsqo5F5",10139]
phase: {"applyExtrinsic":1}, section: balances, method: Deposit, data: ["5GgRLHTmSKBUhqE34DHVgBWZrRbG4xRWLxiCxTPeq37jDHax",10139]
phase: {"applyExtrinsic":1}, section: transactionPayment, method: TransactionFeePaid, data: ["5HHRg2nVfTXD9StGihrjHre5LhC2xVRZhBYvVqoorvsqo5F5",10139,0]
phase: {"applyExtrinsic":1}, section: system, method: ExtrinsicFailed, data: [{"module":{"index":12,"error":"0x07000000"}},{"weight":{"refTime":434338204,"proofSize":0},"class":"Normal","paysFee":"Yes"}]
Lock released
/home/kassem/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:320
            result = await this._applyExtrinsic(extrinsic, resultSections, resultEvents);
                     ^
Error: Failed to apply {"args":{"calls":[{"args":{"contract_id":"30,484"},"method":"cancelContract","section":"smartContractModule"},{"args":{"contract_id":"30,483"},"method":"cancelContract","section":"smartContractModule"},{"args":{"contract_id":"30,484"},"method":"cancelContract","section":"smartContractModule"}]},"method":"batchAll","section":"utility"} due to error: ContractNotExists
    at /home/kassem/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:312:19
    at async TFClient.applyExtrinsic (/home/kassem/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:320:22)
    at async TwinDeploymentHandler.rollback (/home/kassem/tfgrid-sdk-ts/packages/grid_client/src/high_level/twinDeploymentHandler.ts:373:5)
    at async TwinDeploymentHandler.handle (/home/kassem/tfgrid-sdk-ts/packages/grid_client/src/high_level/twinDeploymentHandler.ts:523:7)
    at async GWModule.deploy_name (/home/kassem/tfgrid-sdk-ts/packages/grid_client/src/modules/gateway.ts:82:23)

Omar Kassem, [6/7/23 5:55 PM]

    at async GWModule.descriptor.value (/home/kassem/tfgrid-sdk-ts/packages/grid_client/src/modules/utils.ts:14:12)
    at async GWModule.descriptor.value (/home/kassem/tfgrid-sdk-ts/packages/grid_client/src/helpers/validator.ts:23:12)
    at async main (/home/kassem/tfgrid-sdk-ts/packages/grid_client/scripts/wireguard_vm.ts:97:27)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
➜  grid_client git:(development_add_wireguardExample) ✗
```

**Comment by 0oM4R** [comment]
> I got the correct error after adding some delay before applyAllExtrinsics in grid_client/src/high_level/twinDeploymentHandler.ts

I tried to add the delay but keep getting the same error, can't reach any valid solution yet 
the issue surely in 
https://github.com/threefoldtech/tfgrid-sdk-ts/blob/0ab73db5448ca61f484d836584b94f904f0fb050/packages/grid_client/src/high_level/twinDeploymentHandler.ts#L373

**Comment by mohamedamer453** [comment]
Verified on devnet.

I created a name gateway with the same backend that was mentioned above `10.238.2.2:22`. and it failed with the expected error.
`due to 255 failed to validate backend '10.238.2.2:22': failed to parse backend with error: parse "10.238.2.2:22": first path segment in URL cannot contain colon`

the rollback was also completed successfully and the contract was deleted.

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/518282c7-048d-4208-ab9c-119fda6e6218)
